save.go

```diff
@@ -0,0 +1,138 @@
+package conf
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+)
+
+// SaveTo writes a config struct into the file name in YAML format.
+// name is created if not exists.
+func SaveTo(name string, cfg interface{}) error {
+	f, err := os.Create(name)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	Save(f, cfg)
+	return nil
+}
+
+// Save writes a config struct into w in YAML format.
+func Save(w io.Writer, cfg interface{}) {
+	v := reflect.ValueOf(cfg)
+	if v.Kind() != reflect.Struct {
+		panic(fmt.Sprint("cfg should be a struct"))
+	}
+
+	saveStruct(w, v, 0)
+}
+
+func saveStruct(w io.Writer, v reflect.Value, indent int) {
+	t := v.Type()
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		fv := v.Field(i)
+
+		conf := f.Tag.Get("conf")
+		if conf == "-" {
+			continue
+		}
+
+		confArgs := strings.Split(conf, ",")
+		if len(confArgs) >= 2 && confArgs[1] == "omitempty" {
```

Comment:
This is an extension of the API, currently the `conf` tag doesn't mention it may be comma-separated and support `omitempty`, this needs to be documented.

However, I'm mostly in favor of dropping this, if you want to show what your configuration is you're better off printing out everything.

Do you have an example of when you wouldn't want to see the properties in your configuration?

Also, another issue is the behavior of this `omitempty` tag doesn't match what's done by the standard library => https://golang.org/src/encoding/json/encode.go#L288